tests/MongoDB.Driver.Core.Tests/Core/Authentication/AwsSignatureVersion4Tests.cs

```diff
@@ -0,0 +1,159 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
+using MongoDB.Driver.Core.Authentication;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Tests.Core.Authentication
+{
+    public class AwsSignatureVersion4Tests
+    {
+        [Theory]
+        [InlineData("sts.amazonaws.com", "us-east-1")]
+        [InlineData("first", "us-east-1")]
+        [InlineData("first.second", "second")]
+        [InlineData("first.second.third", "second")]
+        public void GetRegionShouldWorkAsExpected(string host, string expectedRegion)
```

Comment:
nit: rename method to GetRegion_should_return_expected_result

in order to follow are conventions for naming tests

Same for all other methods in this file